dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -129,27 +130,42 @@ public String get(final ContainerRequestContext carrier, final String key) {
     private static class ServerAttributesExtractor
             implements HttpServerAttributesGetter<ContainerRequestContext, ContainerResponseContext> {
 
+
+        private static final String ROUTE = "route";
+
         @Override
         public String flavor(final ContainerRequestContext request) {
             return null;
         }
 
         @Override
         public String route(final ContainerRequestContext request) {
-            Class<?> resourceClass = (Class<?>) request.getProperty(resourceString + "class");
-            Method method = (Method) request.getProperty(resourceString + "method");
 
-            UriBuilder template = UriBuilder.fromResource(resourceClass);
-            String contextRoot = request.getUriInfo().getBaseUri().getPath();
-            if (contextRoot != null) {
-                template.path(contextRoot);
-            }
+            String route = (String) request.getProperty(ROUTE);
+
+            if (route == null) {
+                Class<?> resourceClass = (Class<?>) request.getProperty(resourceString + "class");
+                Method method = (Method) request.getProperty(resourceString + "method");
+
+                UriBuilder template = null;
+                String contextRoot = request.getUriInfo().getBaseUri().getPath();
+
+                if (contextRoot != null) {
```

Comment:
Can this ever be null?

Reply:
I don't see how it can be null, as there should always be a contextRoot of "/" at the minimum, so changed the code. but maybe @yasmin-aumeeruddy knows something I don't?

Reply:
That is a good point. The changes look fine!